champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/bow/NapalmArrow.java

```diff
@@ -0,0 +1,267 @@
+            UtilDamage.doCustomDamage(cde);
+        }
+    }
+
+    @Override
+    public void activate(Player player, int level) {
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_BLAZE_AMBIENT, 2.5F, 2.0F);
+        active.add(player.getUniqueId());
+    }
+
+    @Override
+    public void onHit(Player damager, LivingEntity target, int level) {
+        //ignore
+    }
+
+    @EventHandler
+    public void onArrowDamage(CustomDamageEvent event){
+        if (!(event.getDamager() instanceof Player player)) return;
+        if (!(event.getProjectile() instanceof Arrow arrow)) return;
+        if (!hasSkill(player)) return;
+        if (!napalmArrows.containsValue(arrow)) return;
+
+        int level = getLevel(player);
+        UtilServer.runTaskLater(champions, () -> event.getDamagee().setFireTicks((int) (getBurnDuration(level) * 20)), 1);
+    }
+
+    @EventHandler
+    public void onArrowHit(ProjectileHitEvent event) {
+        if (!(event.getEntity() instanceof Arrow arrow)) return;
+        if (!(arrow.getShooter() instanceof Player player)) return;
+        if (!hasSkill(player)) return;
+        if (!napalmArrows.containsValue(arrow)) return;
+
+        int level = getLevel(player);
+        Location arrowLocation = arrow.getLocation();
+
+        player.getWorld().playSound(arrowLocation, Sound.ENTITY_GENERIC_EXPLODE, 1.0F, 2.0F);
+        doNapalm(player, arrowLocation, level);
+
+        arrow.remove();
+        UtilServer.runTaskLater(champions, () -> napalmArrows.remove(player.getUniqueId()), 1);
+    }
+
+    @Override
+    public void processEntityShootBowEvent(EntityShootBowEvent event, Player player, int level, Arrow arrow) {
+        napalmArrows.put(player.getUniqueId(), arrow);
+        arrow.setFireTicks(200);
+    }
+
+    @UpdateEvent
+    public void updateArrowTrail() {
+        Iterator<Map.Entry<UUID, Arrow>> iterator = napalmArrows.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<UUID, Arrow> entry = iterator.next();
+            Arrow arrow = entry.getValue();
+            if (arrow.isDead() || !arrow.isValid()) {
+                iterator.remove();
+            } else {
+                displayTrail(arrow.getLocation());
+            }
+        }
+    }
+
+    @Override
+    public void displayTrail(Location location) {
+        new ParticleBuilder(Particle.FLAME)
+                .location(location)
+                .count(1)
+                .offset(0.1, 0.1, 0.1)
+                .extra(0)
+                .receivers(60)
+                .spawn();
+    }
+
+    @EventHandler
+    public void onPlayerQuit(PlayerQuitEvent event) {
+        UUID playerId = event.getPlayer().getUniqueId();
+        napalmArrows.remove(playerId);
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent event) {
+        UUID playerId = event.getEntity().getUniqueId();
+        napalmArrows.remove(playerId);
+    }
```

Comment:
Probably wont be necessary with the above changes, arrow will just get removed when it lands or is removed from the servers memory. I feel like you'd still want the napalm shot to go off if the player shot it just before they died anyway